interfaces/builtin/pkcs11.go

```diff
@@ -0,0 +1,166 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016-2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/apparmor"
+	"github.com/snapcore/snapd/interfaces/seccomp"
+	"github.com/snapcore/snapd/snap"
+)
+
+const pkcs11Summary = `allows use of pkcs11 framework and access to exposed tokens`
+
+const pkcs11BaseDeclarationSlots = `
+  pkcs11:
+    allow-installation:
+      slot-snap-type:
+        - core
+        - gadget
+    deny-auto-connection: true
+`
+const pkcs11BaseDeclarationPlugs = `
+  pkcs11:
+    allow-installation: false
+    deny-auto-connection: true
+`
```

Comment:
this is inverted from what we discussed - the pkcs11 plug should not be super-privileged but the slot should be. So you should adjust this to be:

```suggestion
# plugs are okay to be installed but don't auto-connect
const pkcs11BaseDeclarationPlugs = `
  pkcs11:
    allow-installation: true
    allow-auto-connection: false
`

# slots are super-privileged since only gadgets should be exposing them and the pkcs11
# sockets will provide access to device secrets either directly or indirectly thus granting 
# device ownership
const pkcs11BaseDeclarationSlots = `
  pkcs11:
    allow-installation: false
    deny-connection: true
    deny-auto-connection: true
`
```

Reply:
fixed